### PR TITLE
Configurable distance threshold for drag and drop events

### PR DESCRIPTION
**Pull request by ostreifel**
#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [x] Include a change request file if publishing 
- [x] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes
Make the threshold for drag and drop configurable

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by micahgodbolt** [comment]
@ostreifel can you give a little more detail as to the problem/solution. Do we have any documentation you could update to show an example of this?

**Comment by ostreifel** [comment]
@micahgodbolt When I drag an item too slowly the items the onDragStart does not fire because it falls below the threshold of distance for drag

**Comment by micahgodbolt** [comment]
Is there a quick codepen you can use to demonstrate this change? i.e. how can we be sure these changes work?

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.Props.ts [hunk 1]
```diff
@@ -160,6 +160,9 @@ export interface IDetailsListProps extends React.Props<DetailsList> {
 
   /** Boolean value to indicate if the role application should be applied on details list. Set to false by default */
   shouldApplyApplicationRole?: boolean;
+
+  /** The minimum mouse move distance to treat it as drag event */
+  distanceSquaredForDrag?: number;
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.Props.ts]: If we go with this approach, I'd have the user not pass in the squared distance, but just the distance and to any calculation detail ourselves. this leaks an implementation detail. 

We should probably also include that it's pixels? Maybe at least on the comment

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.Props.ts [hunk 2]
```diff
@@ -160,6 +160,9 @@ export interface IDetailsListProps extends React.Props<DetailsList> {
 
   /** Boolean value to indicate if the role application should be applied on details list. Set to false by default */
   shouldApplyApplicationRole?: boolean;
+
+  /** The minimum mouse move distance to treat it as drag event */
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.Props.ts]: what is the @default ?

Comment (ostreifel) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.Props.ts]: Today it's 5 pixels. This allows you to set it to 0.

Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.Props.ts]: can you add that as a `@default` jsdoc comment?

**Pull request merged**